File: 1454B.cpp
#include <bits/stdc++.h>

using namespace std;

#define ar array
#define ll long long

const int MAX_N = 1e5 + 1;
const int MOD = 1e9 + 7;
const int INF = 1e9;
const ll LINF = 1e18;



int solve() {
    int size;
    int ans = -1;
    int min = INF;
    set<int> seen;
    cin >> size;
    vector<int> v1;
    for(int i=0;i<size;i++){
        int temp;
        cin >> temp;
        v1.push_back(temp);
    }

    for(int i=0;i<size;i++){
        if(seen.find(v1[i])==seen.end()){
            seen.insert(v1[i]);
            if(v1[i]<min){
                min = v1[i];
                ans = i+1;
            }
        }
        else{
            min = INF;
            ans = -1;
        }
    }
    return ans;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0); cout.tie(0);
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);

    int tc; cin >> tc;
    for (int t = 1; t <= tc; t++) {
        // cout << "Case #" << t  << ": ";
        int answer = solve();
        cout << answer << endl;
    }
}

File: 1454B.py
from collections import Counter
t = int(input())
for _ in range(t):
    size = int(input())
    li = list(map(int,input().split()))
    c1 = Counter(li)
    ans = []
    answer = -1
    for i,j in c1.items():
        if j>1:
            pass
        else:
            ans.append(i)
    
           
    if ans == []:
        print(-1)
    else:
        a = min(ans)
        for i in range(0,len(li)):
            if li[i] == a:
                answer = i+1 
        print(answer)
